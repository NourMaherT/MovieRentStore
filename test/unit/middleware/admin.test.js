const admin = require('../../../middleware/admin')
const mongoose = require('mongoose');

describe('Admin middleware', () => {
    const scenarios = [
        {
          description: 'should return error 403 if user is not admin',
          error: 'Acsess Denied',
          status: 403
        }
      ];
    scenarios.forEach(({ description, error, status }) => {
        it(description, async () => {
            const user = {
                _id: mongoose.Types.ObjectId().toHexString(),
                isAdmin: false
            };
    
            const req = { user };
            const res = {
                send: jest.fn(),
                status: jest.fn(() => res)
              };
            const next = jest.fn();
    
            await admin(req, res, next);
            expect(res.status).toBeCalledWith(status);
            expect(res.send.mock.calls[0][0]).toMatchSnapshot();
        });
    });

});
