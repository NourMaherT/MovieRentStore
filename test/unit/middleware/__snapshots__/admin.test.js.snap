// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Admin middleware should return error 403 if user is not admin 1`] = `"Acsess Denied"`;
